L1Trigger/L1THGCal/plugins/concentrator/HGCalConcentratorProcessorSelection.cc

```diff
@@ -66,74 +75,67 @@ void HGCalConcentratorProcessorSelection::run(const edm::Handle<l1t::HGCalTrigge
       thickness = HGCalTriggerTools::kScintillatorPseudoThicknessIndex_;
     }
 
-    if (coarsenTriggerCells_ || (fixedDataSizePerHGCROC_ && thickness > kHighDensityThickness_)) {
+    SubDetectorType subdet = hgcal_silicon_CEE;
+    if (thickness != HGCalTriggerTools::kScintillatorPseudoThicknessIndex_ && triggerTools_.isEm(module_trigcell.first) ) subdet = hgcal_silicon_CEE;
+    else if (thickness != HGCalTriggerTools::kScintillatorPseudoThicknessIndex_ && !triggerTools_.isEm(module_trigcell.first) ) subdet = hgcal_silicon_CEH;
+    else if (thickness == HGCalTriggerTools::kScintillatorPseudoThicknessIndex_) subdet = hgcal_scintillator; 
```

Comment:
The subdet should be independent from the thickness. It can be directly retrieved from the detid.

Reply:
Ok - I have changed to using the trigger tools function, but maybe there is a better way.